src/modules/robot/Robot.cpp

```diff
@@ -1461,17 +1461,30 @@ bool Robot::append_arc(Gcode * gcode, const float target[], const float offset[]
     float r_axis1 = -offset[this->plane_axis_1];
     float rt_axis0 = target[this->plane_axis_0] - center_axis0;
     float rt_axis1 = target[this->plane_axis_1] - center_axis1;
+    float angular_travel = 0;
 
-    // Patch from GRBL Firmware - Christoph Baumann 04072015
-    // CCW angle between position and target from circle center. Only one atan2() trig computation required.
-    float angular_travel = atan2f(r_axis0 * rt_axis1 - r_axis1 * rt_axis0, r_axis0 * rt_axis0 + r_axis1 * rt_axis1);
-    if (plane_axis_2 == Y_AXIS) { is_clockwise = !is_clockwise; }  //Math for XZ plane is reverse of other 2 planes
-    if (is_clockwise) { // Correct atan2 output per direction
-        if (angular_travel >= -ARC_ANGULAR_TRAVEL_EPSILON) { angular_travel -= (2 * PI); }
+    //Check to see if we have a full circle, and if so, set angualr_travel.   
+    if ((this->machine_position[this->plane_axis_0]==target[this->plane_axis_0]) and
```

Comment:
please replace and with && as and is not a c++ thing

Reply:
Opps thanks for catching that... I'm surprised I did not get a compiler error.   I've been working on my Pascal project a lot lately, it uses 'And'